### ISSUE TITLE
httplib2 dependency needs to be updated to latest version

### ISSUE THREAD

**Issue by tswast** [google/google-api-python-client#194]
I ran into https://github.com/google/google-api-python-client/issues/135 today when testing https://github.com/GoogleCloudPlatform/cloud-vision/pull/28. Pip installed an old version of httplib2 when I installed the google-api-client-python. Upgrading my local httplib2 version fixed the issue.

```
$ python3 label.py ../../data/label/cat.jpg
Traceback (most recent call last):
  File "label.py", line 78, in <module>
    main(args.image_file)
  File "label.py", line 67, in main
    response = service_request.execute()
  File "/usr/local/lib/python3.4/dist-packages/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/googleapiclient/http.py", line 755, in execute
    method=str(self.method), body=self.body, headers=self.headers)
  File "/usr/local/lib/python3.4/dist-packages/googleapiclient/http.py", line 93, in _retry_request
    resp, content = http.request(uri, method, *args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/oauth2client/client.py", line 622, in new_request
    redirections, connection_type)
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 1120, in request
    headers = self._normalize_headers(headers)
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 1088, in _normalize_headers
    return _normalize_headers(headers)
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 194, in _normalize_headers
    return dict([ (key.lower(), NORMALIZE_SPACE.sub(value, ' ').strip())  for (key, value) in headers.items()])
  File "/usr/lib/python3/dist-packages/httplib2/__init__.py", line 194, in <listcomp>
    return dict([ (key.lower(), NORMALIZE_SPACE.sub(value, ' ').strip())  for (key, value) in headers.items()])
TypeError: sequence item 0: expected str instance, bytes found

$ pip3 install --upgrade google-api-python-client
Requirement already up-to-date: google-api-python-client in /usr/local/lib/python3.4/dist-packages
Cleaning up...

$ sudo pip3 install --upgrade httplib2
[sudo] password for swast: 
Downloading/unpacking httplib2 from https://pypi.python.org/packages/source/h/httplib2/httplib2-0.9.2.tar.gz#md5=bd1b1445b3b2dfa7276b09b1a07b7f0e
  Downloading httplib2-0.9.2.tar.gz (205kB): 205kB downloaded
  Running setup.py (path:/tmp/pip_build_root/httplib2/setup.py) egg_info for package httplib2

Installing collected packages: httplib2
  Found existing installation: httplib2 0.8
    Not uninstalling httplib2 at /usr/lib/python3/dist-packages, owned by OS
  Running setup.py install for httplib2

Successfully installed httplib2
Cleaning up...

$ python3 label.py ../../data/label/cat.jpg      
Found label: cat for ../../data/label/cat.jpg
```

**Comment by tswast** [comment]
@jonparrott Please take a look.

**Comment by nathanielmanistaatgoogle** [comment]
What do you see as the output of `pip3 --version`?

**Comment by jonparrott** [comment]
Also see the discussion over at oauth2client: https://github.com/google/oauth2client/issues/441

**Comment by tseaver** [comment]
Fixed in 6394b660
